extensions/mongoProcessor/mongoProcessorTask.js

```diff
@@ -16,18 +19,82 @@ const healthServer = new HealthProbeServer({
     dump: config.log.dumpLevel });
 
-mongoQueueProcessor.start();
-healthServer.onReadyCheck(log => {
-    if (mongoQueueProcessor.isReady()) {
-        return true;
-    }
-    log.error('MongoQueueProcessor is not ready!');
-    return false;
-});
-healthServer.start();
+const activeProcessors = {};
+
+function initAndStart() {
+    // NOTE: using replication service account
+    const sourceConfig = config.extensions.replication.source;
+    initManagement({
+        serviceName: 'replication',
+        serviceAccount: sourceConfig.auth.account,
+    }, error => {
+        if (error) {
+            log.error('could not load management db', { error });
+            setTimeout(initAndStart, 5000);
+            return;
+        }
+        log.info('management init done');
+
+        let bootstrapList = config.getBootstrapList();
+        config.on('bootstrap-list-update', () => {
+            bootstrapList = config.getBootstrapList();
+
+            const active = Object.keys(activeProcessors);
+            const update = bootstrapList.map(i => i.site);
+            const allSites = [...new Set(active.concat(update))];
+
+            async.each(allSites, (site, next) => {
+                if (update.includes(site)) {
+                    if (!active.includes(site)) {
+                        // new site to setup
+                        const mqp = new MongoQueueProcessor(kafkaConfig,
+                            mongoProcessorConfig, mongoClientConfig, site);
+                        mqp.start();
+                        activeProcessors[site] = mqp;
+                    }
+                } else {
+                    // site no longer in bootstrapList
+                    activeProcessors[site].stop(() => {});
+                    delete activeProcessors[site];
+                }
+                return next();
+            }, err => {
+                if (err) {
+                    process.exit(1);
+                }
+            });
+        });
+
+        // Start Processors for each site
+        const siteNames = bootstrapList.map(i => i.site);
+        async.each(siteNames, (site, next) => {
+            const mqp = new MongoQueueProcessor(kafkaConfig,
+                mongoProcessorConfig, mongoClientConfig, site);
+            mqp.start();
+            activeProcessors[site] = mqp;
+            return next();
+        }, err => {
+            if (err) {
+                process.exit(1);
+            }
+        });
+        healthServer.onReadyCheck(() => {
+            let passed = true;
+            Object.keys(activeProcessors).forEach(site => {
+                if (!activeProcessors[site].isReady()) {
+                    passed = false;
+                    log.error(`MongoQueueProcessor for ${site} is not ready`);
+                }
+            });
+            return passed;
+        });
+        log.info('Starting HealthProbe server');
+        healthServer.start();
+    });
+}
+
+initAndStart();
```

Comment:
Minor: `initAndStart` seems to encompass a lot of different ideas (i.e., initiate the management, update the bootstrap list, start processors, and check the health of the system with the health probe). I think we could improve maintainability and understanding by breaking down into multiple, simpler functions.